app.js

```diff
@@ -13,6 +13,52 @@ function hasFilter(name, filters) {
   return false;
 }
 
+function searchToUrl(search) {
+  if (!search.toString()) {
+    return "";
+  }
+  return "?" + search.toString();
+}
+
+function initialSetStateByUrl() {
```

Comment:
@nt1m I think the way I wrote it, yes it is.

on `setQueryString` function I am using `history.state` object as a mirror of what is querystring filter, this way I can diff the new (param from setQueryString) against the current and based on this remove, add or change values there.

Since I am using `history.state` to keep things sync, without this funciton called on `init`, querystring and `history.state` will be out of sync. When user loads the page with some filter querystring and try to change the values it will call `setQueryString` but it'll cause a bug:

STR:
```
// without initialSetStateByUrl 
- user loads page with querystring "?easy=&mentored=&tool=inspector"
- user clicks on "all" tools
> setQueryString is called, but history.state is empty, so it will produce this new querystring on replaceString "?tool=all" without easy and mentored.
```